cql3/statements/index_prop_defs.cc

```diff
@@ -23,8 +23,8 @@ void cql3::statements::index_prop_defs::validate() {
         throw exceptions::invalid_request_exception("CUSTOM index requires specifying the index class");
     }
 
-    if (!is_custom && custom_class) {
-        throw exceptions::invalid_request_exception("Cannot specify index class for a non-CUSTOM index");
+    if (!is_custom && custom_class && *custom_class != "dummy-vector-backend") {
```

Comment:
I imagine this validation should looks like this:
```
if (!is_custom && custom_class) {
  throw "cannot specify index class for a non-custom index"
}

if (*custom_class in _supported_custom_classes) {
  throw "unsupported custom index class {*custom_class}"
}
```

This way the code will be easier to maintain when we'd like to add a new custom class.

Also, this gives us a better control over supported custom index classes. Most likely we want to protect a newly added index class with a cluster feature (see gms/feature_service.hh), so the node starts using it once all of the nodes are aware of the feature.